src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -73,10 +73,13 @@ object ReadModelQueries {
     producersIds: Seq[String],
     states: Seq[EServiceDescriptorState],
     offset: Int,
-    limit: Int
+    limit: Int,
+    exactMatchOnName: Boolean = false
```

Comment:
I had an idea about this boolean parameter. We may abstract over the query that is used in the aggregation. WDYT @galales?

Reply:
not sure I got what you mean, can you provide an example?

Reply:
passing a boolean parameter that has a default is a bad practice, my idea was to pass the main query object as an argument, wdyt?

Reply:
I agree on the default value for the boolean.

If I get what you mean, I would keep the queries and their implementation inside this helper.
It improves maintainability, especially considering that part of the queries cannot be checked at compile time, at least we know where all the possible impacts on data change may be.
TL;DR single-responsibility principle